src/tools.ts

```diff
@@ -0,0 +1,131 @@
+import fs, { PathLike } from 'fs';
+import { Config } from './config/app';
+import { ComposerJson } from './config/composer';
+import parseJsonFile from './json';
+
+export enum ToolExecutionType {
+    /**
+     * @description Executed on every supported PHP version with lowest & latest dependencies.
+     *              In case, a lock-file is present, the minimum supported PHP version will also run with LOCKED
+     *              dependencies.
+     */
+    MATRIX = 'matrix',
+
+    /**
+     * @description Executed on the minimum PHP version with either LOCKED or LATEST dependencies.
+     */
+    STATIC = 'static',
+}
+
+enum ToolType {
+    LINTER = 'linter',
+    CODE_CHECK = 'code_check',
+}
+
+export type Tool = {
+    executionType: ToolExecutionType,
+    toolType: ToolType,
+    name: string;
+    command: string;
+    filesToCheck: PathLike[]
+}
+
+function detectInfectionCommand(): string {
+    const composerJson: ComposerJson = parseJsonFile('composer.json', true) as ComposerJson;
```

Comment:
Could this not be reused from elsewhere? I expect hitting the filesystem to be the largest bottleneck in this.

Reply:
I don't think that we have to do these kinds of performance improvements and I would prefer not to pass the composer file around the application.

Anyways, thats how it is done right now - we can optimize this somewhere else but for now I would try to focus on the initial idea: improve dev experience when working on our actions.

If we want to do performance improvements, lets do so in dedicated PRs.

Reply:
Fair.